### PR TITLE
chat-list component documentation

### PR DESCRIPTION
**Pull request by seekdavidlee**
This is the documentation for the [ChatList component contribution](https://github.com/microsoftgraph/microsoft-graph-toolkit/pull/2997) to GraphToolKit.

### CODE REVIEW

File: concepts/toolkit/components/chat-list.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+The following example displays a conversation using the `mgt-chat-list` component.
+
+![A screenshot of a chat list component](./images/mgt-chat-list.png)
+
+## Properties
+
+| Attribute                         | Property            | Description                                                                                            |
+| --------------------------------- | ------------------- | ------------------------------------------------------------------------------------------------------ |
+| None                              | chatThreadsPerPage  | The number of chat threads to render initially and subsequent Load more clicks. Required.              |
```
Comment (plasne) [inline concepts/toolkit/components/chat-list.md:31]: Maybe put ticks around "Load more".

File: concepts/toolkit/components/chat-list.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+The `mgt-chat-list` component doesn't define CSS custom properties.
+
+## Events
+
+The following events are fired from the component.
+
+| Event | When is it emitted | Custom data | Cancelable | Bubbles | Works with custom template
+| ------|-------------------|--------------|:-----------:|:---------:|:---------------------------:|
+| `onLoaded` | Fired when all chat threads are loaded. | No | No | No | No |
```
Comment (plasne) [inline concepts/toolkit/components/chat-list.md:45]: Maybe "Fired after the first page of chat threads are loaded."

File: concepts/toolkit/components/chat-list.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+| `onUnselected` | Fired when user selects a different chat thread. | GraphChatThread | No | No | No |
```
Comment (plasne) [inline concepts/toolkit/components/chat-list.md:46]: Maybe "Fired after a user selects a different chat thread if there was one previously selected."

File: concepts/toolkit/components/chat-list.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+| `onSelected` | Fired when user selects a chat thread. | `GraphChat` | No | No | No |
```
Comment (plasne) [inline concepts/toolkit/components/chat-list.md:47]: Maybe "Fired after a user selects a different chat thread."

File: concepts/toolkit/components/chat-list.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+![A screenshot of a chat list component](./images/mgt-chat-list.png)
+
+## Properties
+
+| Attribute                         | Property            | Description                                                                                            |
+| --------------------------------- | ------------------- | ------------------------------------------------------------------------------------------------------ |
+| None                              | chatThreadsPerPage  | The number of chat threads to render initially and subsequent Load more clicks. Required.              |
+| None                              | menuItems           | Array of `ChatListMenuItem` that can be passed into the component to render menu item(s). A `Mark all as read` menu item will be added. When clicked, this button will unbold all chat threads. |
+| None                              | buttonItems         | Array of `ChatListButtonItem` that can be passed into the component to render button(s). |
```
Comment (plasne) [inline concepts/toolkit/components/chat-list.md:33]: These also:

  lastReadTimeInterval?: number;
  selectedChatId?: string;

File: concepts/toolkit/components/chat-list.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+## Events
+
+The following events are fired from the component.
+
+| Event | When is it emitted | Custom data | Cancelable | Bubbles | Works with custom template
+| ------|-------------------|--------------|:-----------:|:---------:|:---------------------------:|
+| `onLoaded` | Fired when all chat threads are loaded. | No | No | No | No |
+| `onUnselected` | Fired when user selects a different chat thread. | GraphChatThread | No | No | No |
+| `onSelected` | Fired when user selects a chat thread. | `GraphChat` | No | No | No |
+| `onMessageReceived` | Fired when a user chat message is receieved. | `ChatMessage` | No | No | No |
```
Comment (plasne) [inline concepts/toolkit/components/chat-list.md:48]: Maybe "Fired after a chat message is received."

File: concepts/toolkit/components/chat-list.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+| `onAllMessagesRead` | Fired when user clicks on `Mark all as read`. | string array of chat id | No | No | No |
```
Comment (plasne) [inline concepts/toolkit/components/chat-list.md:49]: Maybe "Fired after all messages are marked as read as a result of the user selecting that option."

File: concepts/toolkit/components/chat-list.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+The `mgt-chat-list` component doesn't offer templates to override.
+
+## Microsoft Graph permissions
+
+This control uses the following Microsoft Graph APIs and permissions.
+
+| Configuration | Permission | API |
+| - | - | - |
+| User Id is inferred from logged in user. | Chat.Read, Chat.ReadWrite, Chat.ReadBasic | [/users/{userId}/chats/getAllmessages](/graph/teams-changenotifications-chatmessage#subscribe-to-changes-at-the-user-level), [/me/chats](/graph/api/chat-list) |
```
Comment (plasne) [inline concepts/toolkit/components/chat-list.md:61]: It also uses the Teams App permission.
It also uses the People permissions for pics, availability, etc.

**Review by plasne — CHANGES_REQUESTED** [review]

**Comment by Lauragra** [comment]
@seekdavidlee , FYI, this is the public repo for external contributions. Going forward, please use the internal authoring repo for adding new content so that you can follow our established content workflow. Thanks.

File: concepts/toolkit/components/chat-list.md (hunk 8 above)
Comment (seekdavidlee) [reply concepts/toolkit/components/chat-list.md:63]: Do you have the locations of the code so I can document the APIs  that the scopes are used in?

File: concepts/toolkit/components/chat-list.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+The `mgt-chat-list` component doesn't define CSS custom properties.
+
+## Events
+
+The following events are fired from the component.
+
+| Event | When is it emitted | Custom data | Cancelable | Bubbles | Works with custom template
+| ------|-------------------|--------------|:-----------:|:---------:|:---------------------------:|
+| `onLoaded` | Fired after the first page of chat threads are loaded. | No | No | No | No |
```
Comment (andrewDoing) [inline concepts/toolkit/components/chat-list.md:47]: onLoaded fires multiple times, not just once. And I'm PRing a change making it look like
`onLoaded(e: number) => number`
where number is the number of chat threads

File: concepts/toolkit/components/chat-list.md (hunk 8 above)
Comment (seekdavidlee) [reply concepts/toolkit/components/chat-list.md:64]: I found them and updated

File: concepts/toolkit/components/chat-list.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+| `onUnselected` | Fired after a user selects a different chat thread if there was one previously selected. | GraphChatThread | No | No | No |
+| `onSelected` | Fired after a user selects a different chat thread. | `GraphChat` | No | No | No |
+| `onMessageReceived` | Fired after a chat message is received. | `ChatMessage` | No | No | No |
+| `onAllMessagesRead` | Fired after all messages are marked as read as a result of the user selecting the `Mark all as read` option. | string array of chat id | No | No | No |
```
[the remark this answers is not in the document]
Comment (seekdavidlee) [reply concepts/toolkit/components/chat-list.md:47]: Check out the latest order.

File: concepts/toolkit/components/chat-list.md (hunk 9 above)
Comment (seekdavidlee) [reply concepts/toolkit/components/chat-list.md]: Reworded.

**Comment by Lauragra** [comment]
@seekdavidlee , FYI, this is the public mirror for external contributions. For new content, please open your PR in the private authoring repo so you can follow our content workflow.

**Pull request closed without merging**